testcases/OpTestLparFreq.py

```diff
@@ -0,0 +1,156 @@
+                                    (self.system_name, self.powersave_modes_list[i]))
+            time.sleep(5)
+            self.check_mode(self.powersave_check_list[i])
+
+    def ModeFreq_test(self):
+
+        # This test sets Powersave mode to different available modes and
+        # checks whether the frequency chages accordingly.
+
+        self.cv_HMC.run_command("chpwrmgmt -m %s -r sys -o enable -t %s" %
+                                (self.system_name, self.powersave_modes_list[0]))
+        time.sleep(5)
+        static_freq = self.get_freq()
+        self.cv_HMC.run_command("chpwrmgmt -m %s -r sys -o enable -t %s" %
+                                (self.system_name, self.powersave_modes_list[2]))
+        time.sleep(5)
+        max_perf_freq = self.get_freq()
+        self.cv_HMC.run_command("chpwrmgmt -m %s -r sys -o enable -t %s" %
+                                (self.system_name, self.powersave_modes_list[1]))
+        time.sleep(5)
+        dynamic_freq = self.get_freq()
+        print("Average Frequency with Static Powersave mode=%s, Dynamic Performance mode=%s, Maximum Performance mode=%s"
+              % (static_freq, dynamic_freq, max_perf_freq))
+        if dynamic_freq < static_freq:
+            print("Dynamic Performance mode frequency is less than Static Powersave mode frequency")
+            self.failed_test.append("Static Powersave mode frequency check failed")
+        if max_perf_freq < dynamic_freq:
+            print("Maximum Performance mode frequency is less than Dynamic Performance mode frequency")
+            self.failed_test.append("Maximum Performance mode frequency check failed")
+
+    def Lpar_shutdown_freq_test(self):
+
+        # This test Set Powersave mode to "Staic Power Saver" mode when lpar is deactivated
+        # and check if the frequency is set properly when the lpar boots up
+
+        self.cv_HMC.run_command("chsysstate -r lpar -m %s -o shutdown -n %s" %
+                                (self.system_name, self.lpar_name))
+        time.sleep(60)
+        self.cv_HMC.run_command("chpwrmgmt -m %s -r sys -o enable -t %s" %
+                                (self.system_name, self.powersave_modes_list[0]))
+        self.cv_HMC.run_command("chsysstate -r lpar -m %s -o on -b norm -n %s" %
+                                (self.system_name, self.lpar_name))
+        time.sleep(120)
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        self.check_mode(self.powersave_check_list[0])
+
+    def Lpar_Dynamic_freq_test(self):
+
+        # This test Set Powersave mode to "Dynamic Performance Mode" mode and
+        # check if frequency is set based on system utilisation.
+        # ebizzy url must be defined in ~/.op-test-framework.conf
+
+        self.cv_HMC.run_command("chpwrmgmt -m %s -r sys -o enable -t %s" % (self.system_name, self.powersave_modes_list[1]))
+        time.sleep(5)
+        freq_before = self.get_freq()
+        self.console.run_command("cd /root")
```

Comment:
Why not use mkstemp (or similar function) create temporary directory and use it?